### PR TITLE
puddletag: fix for pyparsing 3.1.2

### PR DESCRIPTION
**Pull request by peterhoeg**
## Description of changes

Make it work with pyparsing 3.1.2 ref https://github.com/puddletag/puddletag/issues/884

Also:

1. do not double-wrap 
2. prep for qt6
3. remove outdated comments

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/puddletag/default.nix [hunk 1]
```diff
@@ -43,6 +39,11 @@ python3.pkgs.buildPythonApplication rec {
       --replace share/pixmaps share/icons
   '';
 
+  buildInputs = [
+    qtbase
+    qtwayland
+  ];
```
Comment (dschrempf) [inline pkgs/applications/audio/puddletag/default.nix:45]: Why do we need these now? I think we should have always included `qtbase`, but `qtwayland`?

**Review by dschrempf — APPROVED** [review]

**Comment by peterhoeg** [comment]
Running puddletag in a plasma wayland environment shows this:

> qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""

Adding qtwayland makes the message disappear.

**Comment by dschrempf** [comment]
For reference, I found:
- https://nixos.org/manual/nixpkgs/unstable/#qt-default-nix
- https://github.com/NixOS/nixpkgs/pull/269674

The Nixpkgs manual (first link) actually says we should only activate `qtwayland` on Linux-based systems... but I just checked the Nixpkgs repository and many times, it is just included as is.

**Comment by wegank** [comment]
> The Nixpkgs manual (first link) actually says we should only activate `qtwayland` on Linux-based systems...

Definitely, but it's OK for this package, where `meta.platforms` is set to `platforms.linux`.

**Comment by peterhoeg** [comment]
Any remaining concerns @dschrempf ?

**Comment by dschrempf** [comment]
No! I just tried it out, and it works here. I advice merging this soon; see https://github.com/NixOS/nixpkgs/issues/301857.

**Pull request merged**